src/resources/js/v2/rsvp-manager.js

```diff
@@ -0,0 +1,291 @@
+ *
+ * @since TBD
+ *
+ * @type {PlainObject}
+ */
+tribe.tickets = tribe.tickets || {};
+tribe.tickets.rsvp = tribe.tickets.rsvp || {};
+
+/**
+ * Configures RSVP manager Object in the Global Tribe variable
+ *
+ * @since  TBD
+ *
+ * @type {PlainObject}
+ */
+tribe.tickets.rsvp.manager = {};
+
+/**
+ * Initializes in a Strict env the code that manages the Event Views
+ *
+ * @since  TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} _   Underscore.js
+ * @param  {PlainObject} obj tribe.tickets.rsvp.manager
+ *
+ * @return {void}
+ */
+( function( $, _, obj ) {
+	'use strict';
+	var $document = $( document );
+	var $window = $( window );
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		container: '.tribe-tickets__rsvp-wrapper',
+		loader: '.tribe-common-c-loader',
+		hiddenElement: '.tribe-common-a11y-hidden',
+	};
+
+	/**
+	 * Stores the current ajax request been handled by the manager.
+	 *
+	 * @since TBD
+	 *
+	 * @type {jqXHR|null}
+	 */
+	obj.currentAjaxRequest = null;
+
+	/**
+	 * Containers on the current page that were initialized.
+	 *
+	 * @since TBD
+	 *
+	 * @type {jQuery}
+	 */
+	obj.$containers = $();
+
+	/**
+	 * Saves all the containers in the page into the object.
+	 *
+	 * @since  TBD
+	 *
+	 * @return {void}
+	 */
+	obj.selectContainers = function() {
+		obj.$containers = $( obj.selectors.container );
+	};
+
+	/**
+	 * Clean up the container and event listeners
+	 *
+	 * @since TBD
+	 *
+	 * @param  {jQuery} container Which element we are going to clean up
+	 *
+	 * @return {void}
+	 */
+	obj.cleanup = function( container ) {
+		var $container = $( container );
+
+		$container.trigger( 'beforeCleanup.tribeTicketsRsvp', [ $container ] );
+
+		$container.trigger( 'afterCleanup.tribeTicketsRsvp', [ $container ] );
+	};
+
+	/**
+	 * Setup the container for RSVP management
+	 *
+	 * @since TBD
+	 *
+	 * @todo  Requirement to setup other JS modules after hijacking Click and Submit
+	 *
+	 * @param  {integer}        index     jQuery.each index param
+	 * @param  {Element|jQuery} container Which element we are going to setup
+	 *
+	 * @return {void}
+	 */
+	obj.setup = function( index, container ) {
+		var $container = $( container );
+
+		$container.trigger( 'beforeSetup.tribeTicketsRsvp', [ index, $container ] );
+
+		$container.trigger( 'afterSetup.tribeTicketsRsvp', [ index, $container ] );
+	};
+
+	/**
+	 * Performs an AJAX request.
+	 *
+	 * @since TBD
+	 *
+	 * @param  {object}         data       DOM Event related to the Click action
+	 * @param  {Element|jQuery} $container Which container we are dealing with
+	 *
+	 * @return {void}
+	 */
+	obj.request = function( data, $container ) {
+		var settings = obj.getAjaxSettings( $container );
+
+		// Pass the data received to the $.ajax settings
+		settings.data = data;
+
+		obj.currentAjaxRequest = $.ajax( settings );
```

Comment:
I don't know how often this case will occur, but we don't have a check for `currentAjaxRequest`. If we, for some reason, had more than 1 ajax request going at the same time, this would overwrite this variable. Not that the variable matters as the ajax request would finish anyway, but this is probably an unintended consequence.

Reply:
True! Do you think I should remove it then?